source/common/utility.h

```diff
@@ -54,6 +57,14 @@ class Utility {
    * @param argv forwarded argv argument of the main entry point.
    */
   static void parseCommand(TCLAP::CmdLine& cmd, const int argc, const char* const* argv);
+
+  /**
+   * @param host_port host:port as a string, where host can be IPv4, [IPv6], or a DNS
+   * name.
+   * @return HostAddressType classification of the host address as IPV4, IPV6,
+   * or DNS, or INVALID for basic parse errors
+   */
+  static HostAddressType hostAddressTypeFromHostPort(const std::string& host_port);
```

Comment:
nit: would absl::string_view be more readable here than the string reference?

Reply:
I tried string_view but the RE2::FullMatch() in the implementation didn't understand it.

Also note that in the updated version, we don't attempt to classify the address anymore, only split host:port into host and port.